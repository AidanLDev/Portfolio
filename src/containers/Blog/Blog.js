import React from 'react';
import { Redirect } from 'react-router-dom';
import Icon from '@mdi/react';
import { mdiChevronLeft } from '@mdi/js';

import Button from 'components/Button/Button';

import * as Blogs from 'blogs';
import styles from './Style.module.scss';

const Blog = props => {
  const path = window.location.pathname.split('/').slice(-1)[0];

  const goBack = () => window.history.back();

  const renderSwitch = path => {
    switch (path) {
      case 'deploying-a-website':
        return <Blogs.deployWebsite />;
      case 'must-use-array-methods':
        return <Blogs.javaScriptArrayMethods />;
      case 'using-use-effect':
        return <Blogs.usingUseEffect />;
      case 'quantum-computing':
        return <Blogs.quantumComputing />;
      case 'spread-vs-flatten':
        return <Blogs.spreadFlatten />;
      case 'javascript-prototype':
        return <Blogs.javaScriptProtoType />;
      case 'pending':
        return <Blogs.pending />;
      default:
        return <Redirect to='/blog/search' />;
    }
  };

  return (
    <div className={styles.blogContainer}>
      <div className={styles.buttonWrapper} onClick={() => goBack()}>
        <Icon color='#131211' size={1} path={mdiChevronLeft} />
        <Button>Go Back</Button>
      </div>
      {renderSwitch(path)}
    </div>
  );
};

export default Blog;
