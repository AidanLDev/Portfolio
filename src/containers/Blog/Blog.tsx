import React from 'react';
import { Redirect } from 'react-router-dom';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import Button from '@material-ui/core/Button';

import * as Blogs from 'blogs';
import styles from './Style.module.scss';

const Blog = () => {
  const path: string = window.location.pathname.split('/').slice(-1)[0];

  const goBack = () => window.history.back();

  const renderSwitch = (path: string) => {
    switch (path) {
      // case 'deploying-a-website':
      //   return <Blogs.deployWebsite />;
      // case 'must-use-array-methods':
      //   return <Blogs.javaScriptArrayMethods />;
      // case 'using-use-effect':
      //   return <Blogs.usingUseEffect />;
      // case 'quantum-computing':
      //   return <Blogs.quantumComputing />;
      // case 'spread-vs-flatten':
      //   return <Blogs.spreadFlatten />;
      // case 'javascript-prototype':
      //   return <Blogs.javaScriptProtoType />;
      case 'pending':
        return <Blogs.Pending />;
      default:
        return <Redirect to='/blog/search' />;
    }
  };

  return (
    <div className={styles.blogContainer}>
      <div className={styles.buttonWrapper} onClick={() => goBack()}>
        {/* #131211 */}
        <Button startIcon={<ChevronLeftIcon />}>Go Back</Button>
      </div>
      {renderSwitch(path)}
    </div>
  );
};

export default Blog;
