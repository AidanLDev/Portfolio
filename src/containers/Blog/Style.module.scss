@import 'styles/pallet.scss';
@import 'styles/variables.scss';
@import 'styles/mixins.scss';

.blogContainer {
  background-color: #ffffff;
  width: 70%;
  min-height: 800px;
  margin: auto;
  padding: 10px;
  font-family: medium-content-serif-font, Georgia, Cambria, 'Times New Roman',
    Times, serif;
  p {
    padding: 10px 40px;
    font-size: larger;
  }
  .buttonWrapper {
    cursor: pointer;
    position: absolute;
    top: 85px;
    padding: 20px;
    margin: 0 0 0 50px;
    @media (max-width: $landscapeTablet) {
      display: flex;
      position: inherit;
      margin: 0;
      padding: 0;
    }
  }
  @media (max-width: $portraitTablet) {
    width: 100%;
  }
}
