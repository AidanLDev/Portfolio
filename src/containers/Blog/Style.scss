@import"styles/pallet";
@import"styles/variables";
@import"styles/mixins";

.blogContainer {
  background-color: #ffffff;
  width: 70%;
  min-height: 800px;
  margin: auto;
  padding: 10px;
  font-family: medium-content-serif-font, Georgia, Cambria, "Times New Roman",
    Times, serif;
  p {
    padding: 10px 40px;
    font-size: larger;
  }
  .buttonWrapper {
    cursor: pointer;
    position: absolute;
    top: 85px;
    padding: 20px;
    margin: 0 0 0 50px;
    @media (max-width: 620px) {
      display: flex;
      position: inherit;
      margin: 0;
      width: 90px;
    }
    svg {
      padding: 0;
      left: 5px;
      display: initial;
      position: absolute;
    }
  }
  @media (max-width: $portraitTablet) {
    width: 100%;
  }
}
