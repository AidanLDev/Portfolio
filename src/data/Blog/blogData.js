export const blogs = [
  {
    title: "Blog posts will live here",
    description: "This is just a placeholder",
    route: "pending"
  }
  // {
  //   title: "# Must use JavaScript array methods",
  //   description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Donec adipiscing tristique risus nec feugiat in fermentum. Diam in arcu cursus euismod quis. Facilisis magna etiam tempor orci eu.`,
  //   route: "must-use-array-methods"
  // },
  // {
  //   title: "Reacts useEffect()",
  //   description: `Covering useEffect, converting old life cycle methods to using the React hook`,
  //   route: "using-use-effect"
  // },
  // {
  //   title: "Will Quantum Computing break encryption",
  //   description: `Quantum Comp https://www.youtube.com/watch?v=6H_9l9N3IXU`,
  //   route: "quantum-computing"
  // },
  // {
  //   title: "JS ES6 using Spread vs Flatten",
  //   description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Donec adipiscing tristique risus nec feugiat in fermentum. Diam in arcu cursus euismod quis. Facilisis magna etiam tempor orci eu.`,
  //   route: "spread-vs-flatten"
  // },
  // {
  //   title: "JavaScript prototype",
  //   description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Donec adipiscing tristique risus nec feugiat in fermentum. Diam in arcu cursus euismod quis. Facilisis magna etiam tempor orci eu.`,
  //   route: "javascript-prototype"
  // }
];
