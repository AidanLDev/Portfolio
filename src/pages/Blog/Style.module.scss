@import"styles/pallet.scss";
@import"styles/variables.scss";
@import"styles/mixins.scss";

.container {
  height: 100%;
  background-color: #f7f7f7;
  color: $black;
}
