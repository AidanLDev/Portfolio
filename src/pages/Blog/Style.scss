@import"styles/pallet";
@import"styles/variables";
@import"styles/mixins";

.container {
  height: 100%;
  background-color: #f7f7f7;
  color: $black;
}
