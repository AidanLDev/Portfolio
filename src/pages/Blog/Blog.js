import React from "react";
import Blog from "containers/Blog/Blog";

import styles from "./Style.scss";

const blog = props => {
  return (
    <div className={styles.container}>
      <Blog />
    </div>
  );
};

export default blog;
