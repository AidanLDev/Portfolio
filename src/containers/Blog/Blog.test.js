import React from 'react';
import Blog from './Blog';
import { shallow } from 'enzyme';

it('Renders Blog Component Correctly', () => {
  const component = shallow(<Blog />);
  expect(component).toMatchSnapshot();
});
