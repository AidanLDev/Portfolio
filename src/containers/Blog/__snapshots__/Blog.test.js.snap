// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Blog Component Correctly 1`] = `
<div
  className="blogContainer"
>
  <div
    className="buttonWrapper"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Button))
      startIcon={<Memo />}
    >
      Go Back
    </WithStyles(ForwardRef(Button))>
  </div>
  <Redirect
    to="/blog/search"
  />
</div>
`;
